Added: Auth for all the routes Closes #3

datasciritwik opened the pull request:
Hi, I have added JWT auth for all the routes.

datasciritwik committed:
Added: Auth for all the routes

keshavpj1711 commented:
I will review the changes soon, meanwhile it would be better if you could give a heads up on what you did exactly, also i saw a lot of updates in requirements.txt were they necessary.

datasciritwik commented:
> I will review the changes soon, meanwhile it would be better if you could give a heads up on what you did exactly, also i saw a lot of updates in requirements.txt were they necessary.

Just added JWT for auth for that i have created files 
- user.py contains pydantic models for auth.
- users.py for routing 
- added middleware for handing policies 
- rest just imports and packages updates

keshavpj1711 commented:
Hi @datasciritwik 
thank you for showing interest and taking time to contribute but due to the following reasons i will be closing this PR and i hope this will guide you for future contributions: 
- **No discussion was done before taking up this issue**, resulting in issues with alignment of our goals for the actual feature
-  No backward compatibility as some features might not require protected routes like searching.
- **Dependency bloat**: the pr added 62 new dependencies for a feature which could have been implemented much more minimally
- The auth routes use different patterns than our existing codebase structure.
- Single Large commit, this should have been broken in order to make life simpler for the reviewer

And there were several other issues that i found

### For future contributions: 
- please have a discussion as to what is to be implemented exactly
- include documentation updates
- break large changes to smaller ones: easier to review

thanks for understanding

keshavpj1711 commented on app/routers/users.py:
The auth routes use different patterns than our existing codebase structure.

keshavpj1711 commented on requirements.txt:
dependency bloat

keshavpj1711 reviewed (commented):
please check the comment in the PR
